core/modules/replica/WorkerRequestFactory.h

```diff
@@ -83,124 +80,78 @@ class WorkerRequestFactoryBase {
     /// @return the name of a technology the factory is based upon
     virtual std::string technology() const = 0;
 
-    /**
-     * Create an instance of the replication request
-     *
-     * @see class WorkerReplicationRequest
-     *
-     * @return
-     *   a pointer to the newly created object
-     */
+    /// @see class WorkerReplicationRequest
     virtual WorkerReplicationRequestPtr createReplicationRequest(
             std::string const& worker,
             std::string const& id,
             int priority,
-            std::string const& database,
-            unsigned int chunk,
-            std::string const& sourceWorker) const = 0;
-
-   /**
-    * Create an instance of the replica deletion request
-    *
-    * @see class WorkerDeleteRequest
-    *
-    * @return
-    *   a pointer to the newly created object
-    */
+            WorkerRequest::ExpirationCallbackType const& onExpired,
+            unsigned int requestExpirationIvalSec,
+            ProtocolRequestReplicate const& request) const = 0;
+
```

Comment:
WRT boilerplate, I think you could write a single function templated on `std::shared_ptr<WorkerReplicationRequest>` and `ProtocolRequestReplicate` and get rid of all these `createReplication` variants.

Reply:
Unfortunately, this won't work. I will explain you why in person.